lib/plugins/object.js

```diff
@@ -0,0 +1,138 @@
+const fp = require('fastify-plugin');
+
+const OBJECTS_TABLE = 'warehouse-objects';
+const OBJECT_VARIANTS_TABLE = 'warehouse-object-variants';
+const OBJECTS_HISTORY_TABLE = 'warehouse-objects-history';
+
+module.exports = fp(
+  async function (fastify) {
+    const { dynamo } = fastify;
+
+    fastify.decorate('getObject', async function ({ name, env }) {
+      const params = {
+        Key: { name, env },
+        TableName: OBJECTS_TABLE
+      };
+      const { Item: item } = await dynamo.get(params).promise();
+      return item;
+    });
+
+    fastify.decorate('putObject', function ({
+      name,
+      latestVersion,
+      env,
+      headVersion = null,
+      headTimestamp = null
+    }) {
+      const now = new Date();
+      if (headVersion && !headTimestamp) {
+        headTimestamp = now.getTime();
+      }
+      const lastModified = now.toUTCString();
+      const params = {
+        Key: {
+          name,
+          env,
+          latestVersion,
+          headVersion,
+          headTimestamp,
+          lastModified
+        },
+        TableName: OBJECTS_TABLE
+      };
+      const { Item: item } = await dynamo.get(params).promise();
+      return item;
```

Comment:
Should this be using some other method besides `get`?

Reply:
I do not think so, why?

We have two other methods indeed: one for queryAll and one for batchGet.

Reply:
This is the `putObject` method -- so presumably you'd want to put here instead of getting, right?

Reply:
Sorry, I misread your comment. Yes, should be `put` indeed. Fixed.